packages/frontend/src/ee/features/aiCopilot/components/ClampedTextWithPopover.tsx

```diff
@@ -0,0 +1,70 @@
+import { Box, Button, Popover, ScrollArea, Text } from '@mantine-8/core';
+import { useLayoutEffect, useRef, useState } from 'react';
+
+type ClampedTextWithPopoverProps = {
+    children: string;
+    maxLines?: number;
+    popoverWidth?: number;
+};
+
+const ClampedTextWithPopover = ({
+    children,
+    maxLines = 3,
+    popoverWidth = 400,
+}: ClampedTextWithPopoverProps) => {
+    const textRef = useRef<HTMLParagraphElement>(null);
+    const [isTextClamped, setIsTextClamped] = useState(false);
+
+    useLayoutEffect(() => {
+        if (textRef.current) {
+            const element = textRef.current;
+            setIsTextClamped(element.scrollHeight > element.clientHeight);
+        }
+    }, [children, maxLines]);
```

Comment:
not sure if you can use the hook we already have: `useIsTruncated` 👁️

Reply:
updated! reduce, reuse, recycle ☺️